ferrocene/ci/scripts/mirror-sphinx-shared-resources.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: MIT OR Apache-2.0
+# SPDX-FileCopyrightText: The Ferrocene Developers
+
+
+import os
+import random
+import shlex
+import string
+import subprocess
+import tempfile
+
+
+COMMIT_MSG = "Automated push to `ferrocene/sphinx-shared-resources`"
+MIRROR_DIR = "ferrocene/doc/sphinx-shared-resources"
+MIRROR_REPO = "git@github.com:ferrocene/sphinx-shared-resources.git"
+
+
+# Mirror changes to ferrocene/sphinx-shared-resources.
+#
+# How:
+# 1. Clone ferrocene/sphinx-shared-resources.
+# 2. Copy files from ferrocene/ferrocene subdirectory to cloned repo.
+# 3. Commit and push changes, if there are any.
+#
+# Note that all commands are executed in a temporary directory.
+def mirror_changes():
+    cwd = os.getcwd()
+
+    # clone mirror repository into temporary directory
+    run(f"git clone {MIRROR_REPO} .")
+
+    # create branch
+    rand_id = "".join(random.choices(string.ascii_lowercase + string.digits, k=8))
+    branch_name = f"automation/mirror-sphinx-shared-resources/{rand_id}"
+    run(f"git switch -c {branch_name}")
+
+    # copy contents of mirror directory into mirror repository
+    # `-T` is responsible for only copying the content
+    run(f"cp -frT {cwd}/{MIRROR_DIR} .")
```

Comment:
Copying the files over is probably not the behavior we want, as if a file is deleted in the monorepo it would not be deleted in the mirrored repository.

Given that it'd be better to move this script to the mirrored repo, and we'll want some non-mirrored files there (like a README explaining why there is a mirror, or license files), it would probably be better to keep a list of files to preserve, then delete all the other files, and copy the new files over.

Reply:
> Copying the files over is probably not the behavior we want, as if a file is deleted in the monorepo it would not be deleted in the mirrored repository.

True. I did not consider deletion.

Reply:
> Given that it'd be better to move this script to the mirrored repo, and we'll want some non-mirrored files there (like a README explaining why there is a mirror, or license files), it would probably be better to keep a list of files to preserve, then delete all the other files, and copy the new files over.

Would it be a problem to put those files into the subfolder in `ferrocene/ferrocene`?